js/main.js

```diff
@@ -58,47 +59,54 @@ class InfiniteScroller {
     };
   }
 
-  static getElements(selector) {
-    let splitSelector = selector.split(/(\.|#)/,3)
-    splitSelector.shift()
-    switch(splitSelector[0]){
+  static getElement(selector) {
+
+    // allow dom elements to get passed in directly
+    if(typeof selector === 'object' ) return selector;
+
+    let splitSelector = selector.substring(0, 1);
+    switch(splitSelector){
       case '.':
-        return document.getElementsByClassName(splitSelector[1])
+        return document.getElementsByClassName(selector.substring(1))[0];
+        break;
       case '#':
-        return document.getElementById(splitSelector[1])
+        return document.getElementById(selector.substring(1));
+        break;
       default:
-        throw new Error("The selector must be a class or id, prepended by the identifier ('.'/'#')")
+        throw new Error("The selector must be a class or id, prepended by the identifier ('.'/'#')");
     }
   }
 
   setupContainer() {
-    let parentContainer = this.constructor.getElements(this.selector)[0];
-    parentContainer.className += ' infinite-flex-container-parent-' + this.position;
+
+    let parentContainer = this.constructor.getElement(this.selector);
+    parentContainer.classList.add(`infinite-flex-container-parent-${this.position}`);
```

Comment:
To my shame I had not picked up on this ES6 syntax yet, but I am so happy to see string interpolation.